proposals/lip-implement_extensible_data_persistence_model.md

```diff
@@ -0,0 +1,110 @@
+
+The pattern itself will not impose "FieldSets". It's up to the implementation of an individual entity to provide this feature or not, but no entity will provide granular fields selection.
+
+#### Filters
+
+Using filters provides an easy and powerful way to fetch data from the persistence layer. It's possible to register different filter types when defining entities. Based on the particular filter type, an entity registers few filter names available to be used as JSON objects. Each filter suffix an additional key to the field name, e.g. If you register `address` as a binary filter, an entity will register `address`, `address_eql`, `address_ne` filters and you can pass those filters as JSON object to fetch data. A filter without any suffix to the field name refers by default to `_eql` - equals.
+
+Following is the list of available suffixes based on filter types:
+
+* BOOLEAN (\_eql , \_ne)
+* TEXT (\_eql, \_ne, \_in, \_like)
+* NUMBER (\_eql, \_ne, \_gt, \_gte, \_lt, \_lte, \_in)
+* BINARY (\_eql, \_ne)
+
+If filters are provided as JSON objects, they will always be joined with an `AND` combinator. For instance, specifying filters as `{name: 'Alpha', description_like: 'Bravo'}` results in fetching all results which have a name equal to `Alpha` and description matching `Bravo`. Specifying filters as an array of objects, e.g. `[{name: 'Alpha'}, {description_like: 'Bravo'}]`, will result in joining objects with `OR` combinator, i.e. fetching data which name equal to `Alpha` or description like `Bravo`.
+
+You can register a `CUSTOM` filter, by defining your own key and a function which will return a custom condition.
+
+### Other CRUD Operations 
+We defined the rules only for the read operations above. The remaining of the CRUD operations have no specific rules to restrict. It's up to the implementation of an entity, to implement the logic to persist the objects.
+
+### Future Possibilities
+
+* By having a clear separation of concerns we can focus on performance improvements on the persistence layer entities individually.
+* By having code abstractions and interfaces, we can write and/or replace the storage engine, e.g. to use another database system or database library.
+* By having a centralised persistence layer for accessing data within modules, we can introduce data access authorization, e.g. prevent certain modules from being able to read/write data outside of their responsibility.
+
+## Backward Compatibility
+
+This implementation of this proposal is backwards compatible. The changes only impact the internal persistence logic and aren't causing any changes to the Lisk Core protocol, nor public interfaces.
+
+## Reference Implementation
+
+A reference implementation can be found at:
+https://github.com/LiskHQ/lisk/compare/development...nh-dal
```

Comment:
This link needs to be updated

Reply:
The feature branch is now deleted, I would be in favour of removing `## Reference Implementation` section.

Reply:
I will update the link to https://github.com/LiskHQ/lisk/tree/v1.5.0/storage.